x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/StDistance.java

```diff
@@ -9,45 +9,33 @@
 import org.elasticsearch.xpack.sql.expression.Expression;
 import org.elasticsearch.xpack.sql.expression.Expressions;
 import org.elasticsearch.xpack.sql.expression.FieldAttribute;
-import org.elasticsearch.xpack.sql.expression.function.scalar.BinaryScalarFunction;
 import org.elasticsearch.xpack.sql.expression.gen.pipeline.Pipe;
 import org.elasticsearch.xpack.sql.expression.gen.script.ScriptTemplate;
+import org.elasticsearch.xpack.sql.expression.predicate.BinaryOperator;
+import org.elasticsearch.xpack.sql.expression.predicate.PredicateBiFunction;
 import org.elasticsearch.xpack.sql.tree.NodeInfo;
 import org.elasticsearch.xpack.sql.tree.Source;
 import org.elasticsearch.xpack.sql.type.DataType;
 
 import static org.elasticsearch.xpack.sql.expression.TypeResolutions.isGeo;
-import static org.elasticsearch.xpack.sql.expression.function.scalar.geo.StDistanceProcessor.process;
 import static org.elasticsearch.xpack.sql.expression.gen.script.ParamsBuilder.paramsBuilder;
 
 /**
  * Calculates the distance between two points
  */
-public class StDistance extends BinaryScalarFunction {
+public class StDistance extends BinaryOperator<Object, Object, Double, StDistance.StDistanceFunction> {
```

Comment:
I'd rename it to `StDistanceOperator`

Reply:
`ST_DISTANCE(sh1, sh2)` is not an operator, we just reusing some logic that can be applied to both.

Reply:
Sorry, I had a mistake. My suggestion is to rename `StDistance` to `StDistanceOperation` and `StDistanceFunction` to `StDistance` similarly to other functions/operators in SQL.